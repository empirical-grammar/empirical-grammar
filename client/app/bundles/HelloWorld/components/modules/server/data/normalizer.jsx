'use strict'

 import React from 'react'
 import _ from 'underscore'

 export default function () {

  var _replace = function (obj, oldField, newField, newValue) {
    var newObject = {}
    newObject[newField] = newValue;
    var newObject = _.chain(newObject).merge(obj).omit(oldField).value();
    return newObject
  }

  var _getValue = function (object, oldField) {
    var oldVal, newVal;
    oldVal = object[oldField]
    if (oldVal instanceof Array) {
      newVal = _.pluck(oldVal, 'id');
    } else {
      newVal = oldVal.id;
    }
    return newVal;
  }

  var _normalizeField = function (object, fieldData) {
    var fieldName = fieldData.name;
    var fieldIdName = fieldData.idName;
    var value = _getValue(object, fieldName)
    var newObject = _replace(object, fieldName, fieldIdName, value)
    return newObject
  }


  // PUBLIC
  this.process = function (object, fieldDatas) {
    return _.reduce(fieldDatas, _normalizeField, object);
  }

}
