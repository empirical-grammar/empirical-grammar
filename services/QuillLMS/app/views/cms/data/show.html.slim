.container
  article.activity-form.activity-data.simple-rounded-box
    form.box-full-form.form-vertical data-redirect=cms_activities_path(@activity.classification.key, flag: @activity.flag) action="/"
      .l-section
        h4 Vendor properties
        iframe#activity-iframe scrolling="no" src=@activity.form_url
      button.btn.btn-primary Save