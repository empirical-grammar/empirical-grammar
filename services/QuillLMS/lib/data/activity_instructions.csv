ID,Instructions
1283,
1237,
877,
878,
1224,
443,
449,
1431,
876,
1228,
450,
1236,
1222,
1284,
1195,
1272,
1278,
1275,
1095,
1282,
711,
1096,
1104,
1106,
1109,
1111,
1113,
1114,
1152,
1136,
1151,
1115,
1098,
1116,
1119,
1121,
1123,
1125,
1127,
1129,
1131,
1133,
1134,
1135,
1100,
1156,
1157,
1158,
1159,
1160,
1139,
1140,
1143,
1148,
1149,
1150,
1137,
1102,
775,
433,
844,
846,
645,
650,
717,
441,
438,
439,
440,
858,
1194,
1279,
419,
1221,
898,
890,
424,
897,
899,
780,
426,
1217,
427,
430,
776,
838,
879,
867,
1213,
872,
1214,
866,
1212,
442,
448,
1211,
1220,
994,
999,
420,
792,
990,
859,
613,
429,
428,
779,
1051,
1219,
1196,
841,
840,
1403,
842,
843,
1390,
653,
435,
1004,
857,
1422,
856,
1420,
436,
1000,
862,
434,
1003,
437,
837,
1005,
1001,
868,
869,
985,
986,
987,
1276,
1419,
861,
870,
993,
851,
847,
1418,
1407,
713,
1411,
714,
715,
716,
1409,
1412,
983,
615,
622,
863,
712,
614,
627,
860,
1286,
755,
617,
642,
623,
1415,
756,
586,
606,
853,
854,
1223,
845,
1277,
605,
628,
425,
1290,
432,
902,
752,
1235,
754,
1037,
1048,
880,
882,
1234,
635,
1425,
1426,
600,
634,
633,
852,
1430,
848,
599,
737,
736,
732,
733,
734,
735,
1199,
1203,
1200,
1201,
1205,
1202,
1209,
1210,
1207,
1208,
762,
761,
763,
764,
875,
1002,
640,
1317,
1333,
1280,
757,
1270,
619,
742,
751,
1054,
1055,
418,
781,
995,
431,
777,
1057,
1226,
1225,
769,
768,
766,
767,
773,
772,
1233,
1232,
770,
771,
765,
896,
774,
1215,
1216,
991,
422,
652,
1052,
1359,
1281,
616,
629,
592,
595,
639,
476,
478,
587,
1274,
1273,
585,
901,
584,
641,
1414,
618,
1271,
643,
621,
630,
477,
608,
728,
731,
1285,
1428,
759,
417,
1056,
1429,
884,
620,
729,
730,
638,
783,
632,
474,
475,
473,
1231,
593,
596,
778,
591,
594,
597,
598,
1218,
873,
889,
1427,
888,
992,
447,
1161,
850,
1229,
1230,
849,
273,
138,
140,
1105,
1144,
1107,
1108,
1110,
1112,
1097,
1153,
1145,
1154,
1138,
1117,
1118,
1120,
1122,
1124,
1099,
1126,
1128,
1130,
1132,
1155,
1142,
1141,
1101,
1146,
1147,
1103,
160,
988,
989,
750,
137,
1340,
136,
1368,
285,
126,
1406,
302,
308,
816,
128,
293,
129,
604,
161,
887,
886,
802,
1342,
801,
1392,
181,
804,
1405,
885,
127,
142,
143,
144,
148,
177,
278,
169,
180,
176,
819,
821,
820,
151,
817,
124,
1382,
125,
178,
603,
287,
167,
105,
301,
1320,
147,
179,
152,
636,
155,
96,
288,
164,
168,
163,
162,
174,
283,
145,
107,
1435,
813,
132,
109,
166,
146,
149,
134,
106,
291,
1378,
108,
135,
286,
488,
637,
173,
94,
808,
1386,
809,
1381,
281,
282,
300,
631,
1204,
1206,
172,
271,
154,
803,
1377,
131,
133,
806,
171,
810,
175,
150,
153,
284,
88,
110,
123,
111,
141,
139,
90,
112,
307,
182,
165,
818,
122,
121,
118,
805,
130,
828,
835,
829,
832,
831,
984,
565,
563,
567,
726,
568,
566,
560,
569,
571,
570,
745,
739,
830,
564,
654,
1292,
1287,
583,
574,
581,
572,
575,
576,
559,
577,
743,
738,
827,
573,
655,
727,
561,
578,
580,
582,
744,
562,
579,
834,
833,
648,
13,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
374,Read the passage and correct the underlined errors.
89,Read the passage and correct the spelling of the commonly confused words.
791,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
748,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
15,Read the passage and correct the capitalization errors you find.
611,
51,Read the passage. rewrite the underlined word so it correctly shows a comparison.
7,Read the passage and correct the verb tense errors you find.
251,Read the passage and correct any grammar or punctuation errors you find.
91,Read the passage and correct the the words that don't match the others in the list.
63,Read the passage. Add commas and colons to the lists.
369,Read the passage and correct the underlined errors.
534,Read the passage and correct any grammar or punctuation errors you find.
387,Read the passage and correct any grammar or punctuation errors you find.
19,"Read the passage. rewrite the underlined words as contractions (e.g. doesn't, can't)."
538,Read the passage and correct any grammar or punctuation errors you find.
492,Read the passage and correct any grammar or punctuation errors you find.
254,Read the passage and correct the capitalization errors you find.
362,Read the passage and correct the underlined errors.
378,Read the passage and correct any grammar or punctuation errors you find.
540,Read the passage and correct any grammar or punctuation errors you find.
391,Read the passage and correct any grammar or punctuation errors you find.
311,Read the passage. rewrite the underlined action words so they're in the right tense.
312,Read the passage. rewrite the underlined action words so they're in the right tense.
85,Read the passage and correct the spelling of the commonly confused words.
624,Read the passage. Add commas where needed.
462,Read the passage and correct any grammar or punctuation errors you find.
245,Read the passage and correct the underlined errors.
262,Read the passage and correct the underlined errors.
252,Read the passage and correct the underlined errors.
371,Read the passage and correct any grammar or punctuation errors you find.
881,Read the passage and correct any grammar or punctuation errors you find.
532,Read the passage and correct any grammar or punctuation errors you find.
364,Read the passage and correct the underlined errors.
530,Read the passage and correct any grammar or punctuation errors you find.
499,Read the passage and correct any grammar or punctuation errors you find.
794,Read the passage. rewrite the underlined action words so they're in the right tense.
657,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
382,Read the passage and correct any grammar or punctuation errors you find.
525,Read the passage and correct any grammar or punctuation errors you find.
658,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
406,Read the passage and correct the underlined errors.
367,Read the passage and correct the underlined errors.
75,Read the passage. Add spaces between the quotation marks and the words next to them.
384,Read the passage and correct the underlined errors.
368,Read the passage and correct the underlined errors.
248,"Read the passage. Replace the underlined error with the correct reflexive pronoun (e.g. himself, myself)."
79,Read the passage and correct the verb tense errors you find.
23,Read the passage and correct the verb tense errors you find.
360,Read the passage and correct the underlined errors.
268,Read the passage and correct the underlined errors.
533,Read the passage and correct any grammar or punctuation errors you find.
77,Read the passage. Find the places where two describing words are connected by and. Replace and with a comma to practice one way of using describing words. Remember to check your spacing!
47,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
398,Read the passage and correct the underlined errors.
53,Read the passage. Add commas to the addresses.
233,Read the passage and correct any grammar or punctuation errors you find.
797,Read the passage. rewrite the underlined action words so they're in the right tense.
370,Read the passage and correct the underlined errors.
260,Read the passage and correct any grammar or punctuation errors you find.
405,Read the passage and correct the underlined errors.
381,Read the passage and correct the underlined errors.
457,Read the passage and correct any grammar or punctuation errors you find.
545,Read the passage and correct any grammar or punctuation errors you find.
394,Read the passage and correct any grammar or punctuation errors you find.
9,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
542,Read the passage and correct any grammar or punctuation errors you find.
544,Read the passage and correct any grammar or punctuation errors you find.
264,Read the passage and correct the underlined errors.
535,Read the passage and correct any grammar or punctuation errors you find.
383,Read the passage and correct any grammar or punctuation errors you find.
400,Read the passage and correct any grammar or punctuation errors you find.
403,Read the passage and correct any grammar or punctuation errors you find.
361,Read the passage and correct the underlined errors.
267,Read the passage and correct the underlined errors.
263,Read the passage and correct the underlined errors.
266,Read the passage and correct the underlined errors.
524,Read the passage and correct any grammar or punctuation errors you find.
527,Read the passage and correct any grammar or punctuation errors you find.
626,Read the passage and correct any grammar or punctuation errors you find.
386,Read the passage and correct any grammar or punctuation errors you find.
41,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
35,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
407,Read the passage and correct the underlined errors.
393,Read the passage and correct any grammar or punctuation errors you find.
539,Read the passage and correct any grammar or punctuation errors you find.
723,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
724,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
536,Read the passage and correct any grammar or punctuation errors you find.
397,Read the passage and correct the underlined errors.
249,Read the passage and correct the underlined errors.
517,Read the passage and correct any grammar or punctuation errors you find.
537,Read the passage and correct any grammar or punctuation errors you find.
358,Read the passage and correct any grammar or punctuation errors you find.
253,Read the passage. Add commas to the lists and the dates.
503,Read the passage and correct any grammar or punctuation errors you find.
518,Read the passage and correct any grammar or punctuation errors you find.
408,Read the passage and correct the underlined errors.
244,Read the passage and correct any grammar or punctuation errors you find.
549,Read the passage and correct any grammar or punctuation errors you find.
363,Read the passage and correct the underlined errors.
409,Read the passage and correct the underlined errors.
69,Read the passage. Add commas where needed after the introductory words.
747,Read the passage and correct any grammar or punctuation errors you find.
725,"Read the passage. rewrite the underlined words so they are in the future perfect tense. Form the future perfect by using ""will have"" and a special past tense form of the action word. "
543,Read the passage and correct any grammar or punctuation errors you find.
531,Read the passage and correct any grammar or punctuation errors you find.
401,Read the passage and correct any grammar or punctuation errors you find.
455,Read the passage and correct any grammar or punctuation errors you find.
37,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
376,Read the passage and correct the underlined errors.
388,Read the passage and correct the underlined errors.
504,Read the passage and correct any grammar or punctuation errors you find.
21,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
11,Read the passage and correct the spelling of the plural words.
315,Read the passage and correct any grammar or punctuation errors you find.
379,Read the passage and correct any grammar or punctuation errors you find.
255,Read the passage and correct the underlined errors.
399,Read the passage and correct any grammar or punctuation errors you find.
101,Read the passage and correct any grammar or punctuation errors you find.
490,Read the passage and correct any grammar or punctuation errors you find.
256,Read the passage and correct any grammar or punctuation errors you find.
392,Read the passage and correct any grammar or punctuation errors you find.
411,Read the passage and correct the underlined errors.
710,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
261,Read the passage and correct any grammar or punctuation errors you find.
464,Read the passage and correct any grammar or punctuation errors you find.
365,Read the passage and correct the underlined errors.
412,Read the passage and correct the underlined errors.
39,Read the passage and correct the capitalization errors you find.
95,Read the passage and correct the spelling of the commonly confused words.
380,Read the passage and correct any grammar or punctuation errors you find.
377,Read the passage and correct the underlined errors.
460,Read the passage and correct any grammar or punctuation errors you find.
719,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
609,Read the passage and correct any grammar or punctuation errors you find.
523,Read the passage and correct any grammar or punctuation errors you find.
520,Read the passage and correct any grammar or punctuation errors you find.
547,Read the passage and correct any grammar or punctuation errors you find.
481,Read the passage and correct any grammar or punctuation errors you find.
516,Read the passage and correct any grammar or punctuation errors you find.
548,Read the passage and correct any grammar or punctuation errors you find.
389,Read the passage and correct any grammar or punctuation errors you find.
345,Read the passage and correct any grammar or punctuation errors you find.
522,Read the passage and correct any grammar or punctuation errors you find.
269,Read the passage and correct the the words that don't match the others in the list.
366,Read the passage and correct the underlined errors.
489,Read the passage and correct any grammar or punctuation errors you find.
309,Read the passage and correct the underlined errors.
43,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
55,Read the passage. Add quotation marks and punctuation to the dialogue.
505,Read the passage and correct any grammar or punctuation errors you find.
491,Read the passage and correct any grammar or punctuation errors you find.
265,Read the passage and correct the underlined errors.
402,Read the passage and correct any grammar or punctuation errors you find.
528,Read the passage and correct any grammar or punctuation errors you find.
798,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
529,Read the passage and correct any grammar or punctuation errors you find.
793,Read the passage. rewrite the underlined word so it correctly shows a comparison.
521,Read the passage and correct any grammar or punctuation errors you find.
215,Read the passage and correct any grammar or punctuation errors you find.
541,Read the passage and correct any grammar or punctuation errors you find.
795,"Read the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/). "
546,Read the passage and correct any grammar or punctuation errors you find.
93,Read the passage and correct the spelling of the commonly confused words.
550,Read the passage and correct any grammar or punctuation errors you find.
113,