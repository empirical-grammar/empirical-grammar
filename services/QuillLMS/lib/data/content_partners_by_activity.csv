Tool,UID,Activity ID,Content Partner,Activity Name
Connect,-LvGsoTdi-HFa4r9eNgK,1095,College Board,“The Red Fox Fur Coat”
Connect,-M-H0DCwYDcD3OsfuXWY,1195,College Board,"""The First Day"""
Connect,-M03YCpAMJn8IxSPZKPa,1222,College Board,"""An Occurrence at Owl Creek Bridge"""
Connect,-M03YpZB2gHlSsxS542O,1223,College Board,Lamb to the Slaughter
Connect,-M03WbwMR1ZaXMAFLXro,1231,College Board,What Happened During the Ice Storm
Connect,ffb1d005-ff57-4a02-ba25-c744ef795b4c,1271,College Board,The Night Circus
Connect,39da04f2-eae9-456d-a255-a36bf0f565d0,1272,College Board,"“Drowning in Dishes, but Finding a Home”"
Connect,2c4518d6-2417-479c-8c6a-782e7692ed41,1273,College Board,The Hamilton Mixtape
Connect,84235cb1-bd7a-4404-a2ac-a47dcf7f23cd,1274,College Board,The Girl Who Fell from the Sky
Connect,b03713db-bced-4890-81e8-6414a0dcf46e,1275,College Board,“The Decline of the American Teenager’s Summer Job”
Connect,39239b1f-7caa-4703-b9ca-d90778eb4872,1277,College Board,Lottery
Connect,ab9f51d3-ccf4-49fc-b621-b30721f5583b,1278,College Board,“Teenagers have Stopped Getting Summer Jobs—Why?”
Connect,bc61293f-b376-437c-bc48-e6b98d4b0d81,1279,College Board,All the Light We Cannot See
Connect,7712a90e-8dd7-4297-bf26-b9747b2a2f86,1280,College Board,Romeo and Juliet
Connect,21eac116-a51a-4ea0-8ce6-ff7aa576fe04,1281,College Board,Tamara's Opus
Connect,2e04ef36-ceea-4544-b7c7-751f89fc2b34,1282,College Board,“What to Do with Kids This Summer? Put ‘Em to Work”
Connect,-M1WgGWcOk22p63fohFK,1283,College Board,1984
Connect,-M1WgDhT2QMzj_vnImab,1284,College Board,"""The Fight"""
Connect,-M1WgLi9ZmrvpyXNoshp,1285,College Board,"The Work You Do, the Person You Are"
Connect,6b69f10b-8d7b-45df-8558-dc3da0a5f140,1290,College Board,Out of My Mind
Connect,-Kbk-mjzTNeqWbvVG9zI,476,Core Knowledge,The Boston Massacre
Connect,-KbvuSGyq6ANjB6cSzpQ,477,Core Knowledge,The Stamp Act
Connect,-Kbvv3WS6vGNkeW7d6dt,478,Core Knowledge,The Boston Tea Party
Proofreader,-Kc-iJqt0TsWgfwYkT6m,481,Core Knowledge,The Sons of Liberty
Proofreader,-KoJ1PFy2DNaRhEtlWex,499,Core Knowledge,George Washington
Proofreader,-KoJ9Lrs-pUNFrzjFTz6,503,Core Knowledge,Sailing with the Stars
Proofreader,-KoJB86Hm3C2_OOo9zA6,504,Core Knowledge,The Cape of Good Hope
Proofreader,-KoJCM7vJTVowwUG_NpH,505,Core Knowledge,Totem Poles
Proofreader,-KqUMM-Akp6DDQlVlAbq,516,Core Knowledge,Spices
Proofreader,-KqUQCCoHDkIggc6BMtU,517,Core Knowledge,Roanoke
Proofreader,-KqU.S.2G2QYq3KWYPHazn,518,Core Knowledge,Santa Maria del Fiore
Proofreader,-KqUXZEm8IdwvR2frMfO,520,Core Knowledge,The School of Athens
Proofreader,-KqYDjR0nWR2gNhyOwWY,521,Core Knowledge,Who was Cervantes?
Proofreader,-KqYHrOHtQSIQ4B2T-wa,523,Core Knowledge,The Sapa Inca
Proofreader,-KqYJAxtdM6HVQi7vHpj,524,Core Knowledge,Native Americans & Horses
Proofreader,-KqYKnytOMErPTLZAgUl,525,Core Knowledge,Godasiyo the Woman Chief
Proofreader,-KqYOlcdRBPIkG6Fe-oc,527,Core Knowledge,Navajo Code Talkers
Proofreader,-KqYRDChl1VQ_QB7rEUL,528,Core Knowledge,Vision Quests
Proofreader,-KqYSjSU.S.SUnLNETJEzg,529,Core Knowledge,What's in a Bean?
Proofreader,-KqYXnbNX5-yKYBF0MHF,530,Core Knowledge,Fruit or Vegetable?
Proofreader,-KqZ1GcvRyn-0x5ph2KS,531,Core Knowledge,Tenochtitlán
Proofreader,-KqdKIZmtEyLE5TTfmwp,532,Core Knowledge,Fossils
Proofreader,-KqdMzxI9fqItxVh_Vgy,533,Core Knowledge,Isabella I of Castile
Proofreader,-KqdjKv54k5iifw2FzDk,534,Core Knowledge,Plants: Bee Gardens
Proofreader,-Kqdo2aBDw7bJ4YIXynn,535,Core Knowledge,Maya Calendar
Proofreader,-KqhXp64K1DKX5EtaSYG,536,Core Knowledge,Plate Tectonics
Proofreader,-KqhpFWpVQubJhqWrkFb,537,Core Knowledge,Rocks
Proofreader,-Kqipm5LV2GUkQYGQ8Lh,538,Core Knowledge,Cactus Photosynthesis
Proofreader,-Kqir7E9_3JftVmWhm7z,539,Core Knowledge,Paul Revere's Ride
Proofreader,-KqixRPyWpTzXtBmZtmm,540,Core Knowledge,Colonial Life in the American Revolution
Proofreader,-Kr1Tc2R9sunf8yHM5xh,541,Core Knowledge,Women in the American Revolution
Proofreader,-Kr1VzAFLsATQbuRJlHd,542,Core Knowledge,Man on the Moon
Proofreader,-Kr1cS7S74s-RHF28y9H,544,Core Knowledge,Mars
Proofreader,-Kr20EDdSpxQm93BJrco,545,Core Knowledge,Mae Jemison
Proofreader,-Kr23P_QLluLsDOeT7MB,547,Core Knowledge,The Silk Road
Proofreader,-Kr24ZZ6uu51bYXhADhG,549,Core Knowledge,Shihuangdi's Search for Immortality
Proofreader,-Kr26YEA35YGuXWE0jzb,550,Core Knowledge,Zheng He and the Pirate
Connect,-KqZdYHq5N9O1OYfzZbl,584,Core Knowledge,The Iroquois Confederacy
Connect,-KqZWcwJndWJBU-U3ehj,585,Core Knowledge,The Importance of the Buffalo
Connect,-KqZh2mwDeSPLCxp2UAT,586,Core Knowledge,Ishi
Connect,-KqZaKPdDL_-tnCUDIlX,587,Core Knowledge,The Dawes Act
Connect,-KqZiV1bnPe48h6gyYss,605,Core Knowledge,Magellan
Connect,-KqcJOzHDYUrioJ6fPlR,606,Core Knowledge,Jamestown
Connect,-KqZ_8tESejeyBmj-_pO,608,Core Knowledge,The Travels of Marco Polo
Connect,-KqZUaz7rINZ8fNssJ6r,613,Core Knowledge,Asteroids
Connect,-KqZR__u2hpJ2z8cIch-,614,Core Knowledge,Ham and Alan Shepard
Connect,-KqZFTXwQ5MCZMH9-bUN,615,Core Knowledge,Eclipses
Connect,-KqiiQTci2gs9zhx2tIA,616,Core Knowledge,Tang Dynasty Inventions
Connect,-KqilM2xWovC-Hudmvjn,617,Core Knowledge,Hui Zong's Defeat
Connect,-Kqisu7eUfGfGvjRvT2W,618,Core Knowledge,The Mongol Invasion of China
Connect,-KqigC-G1kUp_pEeQJjl,619,Core Knowledge,Shihuangdi Unifies China
Connect,-KqYkktoWRhHby2187G8,621,Core Knowledge,The Rock Cycle
Connect,-KqYgUp3xAAJYxZtX2yr,622,Core Knowledge,Fold Mountains
Connect,-KqYd2Nk25H2e3fdkwcs,623,Core Knowledge,Hydrothermal Vents
Connect,-KqcjgL00W0ro-Wr4Bts,627,Core Knowledge,Hernan Cortes
Connect,-KqcdV5JA9uU.S.Y09VvOP,628,Core Knowledge,Maya Mystery
Proofreader,-KqZ1GcvRyn-0x5ph2KS,629,Core Knowledge,Tenochtitlán
Connect,-KqcTJxEiuq07PJzbDIc,630,Core Knowledge,The Royal Road
Connect,-KqZQ1fZyO-Lz5UwooN2,632,Core Knowledge,Vascular and Non-vascular Plants
Connect,-KqZLtzic921c7eDVc1D,633,Core Knowledge,Plants in the Food Chain
Connect,-KqZIkV8V3_92A2ajc4u,634,Core Knowledge,Plant Cells and Animal Cells
Connect,-KqZECh5ZxntzLDB-xQK,635,Core Knowledge,Parts of Vascular Plants
Connect,-KqZpAZ8UMhwZp_3ezqw,640,Core Knowledge,Renaissance Beginnings in Italy
Connect,-KqZvArG8422vhwB7rIo,641,Core Knowledge,The Medici Family
Connect,-KqZldiOXNWbJNFvIRY-,642,Core Knowledge,Humanism
Connect,-KqdGWI1rgvtQBg2vIb0,643,Core Knowledge,The Northern Renaissance
Connect,-KqY7BQmhDn71y-E-s3C,728,Core Knowledge,The War Begins at Lexington and Concord
Connect,-KqYNoENpFXSttDu-xPf,729,Core Knowledge,Turning Point: Bunker Hill
Connect,-KqYPpC3o3JeeuhQadBF,730,Core Knowledge,Turning Point: Saratoga
Connect,-KqYK5A0fNos0ZbUjjOu,731,Core Knowledge,The War Ends at Yorktown
Proofreader,-L8ZGcyIM49GOsX_uOPN,747,Core Knowledge,Sputnik 1
Connect,-LAo-erEroO6r6XV1tAV,756,Core Knowledge,International Space Station
Connect,-KhO2hv2Wl8rVCHp1B1G,860,Core Knowledge,Hernando de Soto