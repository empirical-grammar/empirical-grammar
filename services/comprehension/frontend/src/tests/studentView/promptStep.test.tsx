import * as React from 'react'
import { shallow } from 'enzyme';

import PromptStep from '../../components/studentView/promptStep'
import EditorContainer from '../../components/studentView/editorContainer'

import { activityOne, optimalSubmittedResponse, suboptimalSubmittedResponse, } from './data'

const prompt = activityOne.prompts[0]

const defaultProps = {
  active: false,
  className: 'step',
  everyOtherStepCompleted: false,
  submitResponse: () => {},
  completeStep: () => {},
  stepNumberComponent: <span />,
  onClick: () => {},
  prompt,
  passedRef: () => {},
  submittedResponses: []
}

describe('PromptStep component', () => {
  describe('inactive state', () => {
    const wrapper = shallow(<PromptStep
      { ...defaultProps}
    />)

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('active state', () => {
    describe('before any responses have been submitted', () => {
      const wrapper = shallow(<PromptStep
        { ...defaultProps}
        active
        className="step active"
      />)

      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('renders an EditorContainer', () => {
        expect(wrapper.find(EditorContainer)).toHaveLength(1)
      })

      it('has a button with a disabled class and the text "Get feedback"', () => {
        expect(wrapper.find('button.disabled')).toHaveLength(1)
        expect(wrapper.find('button.disabled').text()).toEqual("Get feedback")
      })

    })

    describe('before any responses have been submitted', () => {
      const wrapper = shallow(<PromptStep
        { ...defaultProps}
        active
        className="step active"
      />)

      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('renders an EditorContainer', () => {
        expect(wrapper.find(EditorContainer)).toHaveLength(1)
      })

      it('has a button with a disabled class and the text "Get feedback"', () => {
        expect(wrapper.find('button.disabled')).toHaveLength(1)
        expect(wrapper.find('button.disabled').text()).toEqual("Get feedback")
      })

    })

    describe('when an optimal response has been submitted', () => {
      const submittedResponses = [optimalSubmittedResponse]
      const wrapper = shallow(<PromptStep
        { ...defaultProps}
        active
        className="step active"
        submittedResponses={submittedResponses}
      />)

      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('renders an EditorContainer', () => {
        expect(wrapper.find(EditorContainer)).toHaveLength(1)
      })

      it('has a non-disabled button with the text "Start next sentence"', () => {
        expect(wrapper.find('button.disabled')).toHaveLength(0)
        expect(wrapper.find('button').text()).toEqual("Start next sentence")
      })

    })

    describe('when a suboptimal response has been submitted', () => {
      const submittedResponses = [suboptimalSubmittedResponse]
      const wrapper = shallow(<PromptStep
        { ...defaultProps}
        active
        className="step active"
        submittedResponses={submittedResponses}
      />)

      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('renders an EditorContainer', () => {
        expect(wrapper.find(EditorContainer)).toHaveLength(1)
      })

      it('has a disabled button with the text "Get new feedback"', () => {
        expect(wrapper.find('button.disabled')).toHaveLength(1)
        expect(wrapper.find('button.disabled').text()).toEqual("Get new feedback")
      })
    })

    describe('when the max attempts have been reached', () => {
      const submittedResponses = [suboptimalSubmittedResponse, suboptimalSubmittedResponse, suboptimalSubmittedResponse, suboptimalSubmittedResponse, suboptimalSubmittedResponse]
      const wrapper = shallow(<PromptStep
        { ...defaultProps}
        active
        className="step active"
        submittedResponses={submittedResponses}
      />)

      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('renders an EditorContainer', () => {
        expect(wrapper.find(EditorContainer)).toHaveLength(1)
      })

      it('has a non-disabled button with the text "Start next sentence"', () => {
        expect(wrapper.find('button.disabled')).toHaveLength(0)
        expect(wrapper.find('button.disabled').text()).toEqual("Start next sentence")
      })
    })

    describe('when the max attempts have been reached or the last answer is optimal and every other question has been completed', () => {
      const submittedResponses = [optimalSubmittedResponse]
      const wrapper = shallow(<PromptStep
        { ...defaultProps}
        active
        everyOtherStepCompleted
        className="step active"
        submittedResponses={submittedResponses}
      />)

      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('renders an EditorContainer', () => {
        expect(wrapper.find(EditorContainer)).toHaveLength(1)
      })

      it('has a non-disabled button with the text "Done"', () => {
        expect(wrapper.find('button.disabled')).toHaveLength(0)
        expect(wrapper.find('button.disabled').text()).toEqual("Done")
      })
    })

  })

})
