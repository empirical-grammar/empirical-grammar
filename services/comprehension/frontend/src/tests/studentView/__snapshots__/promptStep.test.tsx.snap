// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromptStep component active state before any responses have been submitted matches snapshot 1`] = `ShallowWrapper {}`;

exports[`PromptStep component active state before any responses have been submitted matches snapshot 2`] = `ShallowWrapper {}`;

exports[`PromptStep component active state when a suboptimal response has been submitted matches snapshot 1`] = `ShallowWrapper {}`;

exports[`PromptStep component active state when an optimal response has been submitted matches snapshot 1`] = `ShallowWrapper {}`;

exports[`PromptStep component active state when the max attempts have been reached matches snapshot 1`] = `ShallowWrapper {}`;

exports[`PromptStep component active state when the max attempts have been reached or the last answer is optimal and every other question has been completed matches snapshot 1`] = `ShallowWrapper {}`;

exports[`PromptStep component inactive state renders 1`] = `ShallowWrapper {}`;
